tests/System.Numerics.Tensors.Tests/TensorTests.cs

```diff
@@ -0,0 +1,1490 @@
+            Assert.Equal(0, defaultTensor.Length);
+            Assert.Equal(0, defaultTensor.Rank);
+
+            // don't throw
+            var clone = defaultTensor.Clone();
+            clone = defaultTensor.CloneEmpty();
+            var doubleClone = defaultTensor.CloneEmpty<double>();
+            var arr = new int[0];
+            defaultTensor.CopyTo(arr, 0);
+            defaultTensor.Fill(0);
+            defaultTensor.Reshape();
+            defaultTensor.ReshapeCopy();
+            defaultTensor.ToString();
+
+            Assert.Equal(default(Tensor<int>), defaultTensor);
+            Assert.Equal(default(Tensor<int>).GetHashCode(), defaultTensor.GetHashCode());
+            Assert.False(defaultTensor.GetEnumerator().MoveNext());
+
+            // rank is too small
+            Assert.Throws<InvalidOperationException>(() => defaultTensor.GetDiagonal());
+            Assert.Throws<InvalidOperationException>(() => defaultTensor.GetTriangle());
+            Assert.Throws<InvalidOperationException>(() => defaultTensor.GetUpperTriangle());
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => defaultTensor[0]);
+            Assert.Throws<ArgumentOutOfRangeException>(() => defaultTensor[0, 0]);
+            Assert.Throws<ArgumentOutOfRangeException>(() => defaultTensor[0, 0, 0]);
+
+            Assert.Throws<ArgumentException>(() => defaultTensor + 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor + defaultTensor);
+            Assert.Throws<ArgumentException>(() => +defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor - 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor - defaultTensor);
+            Assert.Throws<ArgumentException>(() => -defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor++);
+            Assert.Throws<ArgumentException>(() => defaultTensor--);
+            Assert.Throws<ArgumentException>(() => defaultTensor * 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor * defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor / 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor / defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor % 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor % defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor & defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor & -1);
+            Assert.Throws<ArgumentException>(() => defaultTensor | defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor | -1);
+            Assert.Throws<ArgumentException>(() => defaultTensor ^ defaultTensor);
+            Assert.Throws<ArgumentException>(() => defaultTensor ^ -1);
+            Assert.Throws<ArgumentException>(() => defaultTensor >> 1);
+            Assert.Throws<ArgumentException>(() => defaultTensor << 1);
```

Comment:
Sure I can polish these up.  These aren't really meant to be a full suite of tests yet but merely a start.